### DISCUSSION TITLE
Concerns of FunctionsPreservedDependencies

### DISCUSSION THREAD

**Discussion by SuyliuMS** [Azure/Azure-Functions#2348] [Polls]
Hi, team,

We are going to use OpenTelemetry 1.4.0 in our function, but OpenTelemetry has a dependency to System.Diagnostics.DiagnosticSource and it will throw a exception when trigger functions:
FileLoadException: Could not load file or assembly 'System.Diagnostics.DiagnosticSource, Version=7.0.0.0

However we can add FunctionsPreservedDependencies  element to .csproj file to fix this error:
```
  <ItemGroup>
    <FunctionsPreservedDependencies Include="System.Diagnostics.DiagnosticSource.dll" />
  </ItemGroup>
```

**Will System.Diagnostics.DiagnosticSource.dll in FunctionsPreservedDependencies affect function behaviors?**

We are basically wondering what would be the impact for preserving our own dependency version for dlls also used by the function host and if it’s ok to preserve the System.Diagnostics.DiagnosticSource. We are on dedicated plan and we don’t use AppInsight – all of our instrumentations are Geneva based.

**Comment by fabiocav** [comment]
@SuyliuMS closing stale discussions, the current recommendation is to move to the .NET Isolated model, where this won't be a requirement https://learn.microsoft.com/en-us/azure/azure-functions/migrate-dotnet-to-isolated-model?tabs=net8
